components/calendar/src/types.rs

```diff
@@ -80,6 +80,23 @@ pub struct WeekOfMonth(pub u32);
 #[derive(Clone, Copy, Debug, PartialEq)]
 pub struct WeekOfYear(pub u32);
 
+/// A day of week in month. 1-based.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct DayOfWeekInMonth(pub u32);
```

Comment:
`question`: why not `u8`?

Reply:
Just templating from the other types defined in the file: neither DayOfMonth nor WeekOfMonth need more than a u8 but are also u32s. I'm happy to scope it down to u8 if desired.

Reply:
@Manishearth @sffc do you have opinions here?